sdk/Dockerfile

```diff
@@ -1,78 +1,68 @@
+# syntax=docker/dockerfile:1
+
 # SPDX-FileCopyrightText: Copyright (c) 2023 Intel Corporation
-#
 # SPDX-License-Identifier: BSD-3-Clause
 
-# Use the specific version of ubuntu image
-FROM ubuntu:22.04@sha256:2b7412e6465c3c7fc5bb21d3e6f1917c167358449fecac8176c6e496e5c1f05f as builder
+ARG IMAGE_CACHE_REGISTRY=docker.io
+ARG IMAGE_NAME=library/ubuntu:22.04@sha256:a6d2b38300ce017add71440577d5b0a90460d0e57fd7aec21dd0d1b0761bbfb2
 
-# Set the maintainer label
-LABEL maintainer="qiang.han@intel.com"
+FROM ${IMAGE_CACHE_REGISTRY}/${IMAGE_NAME} as builder
 
-# Set environment variables
 ENV DEBIAN_FRONTEND=noninteractive
+ENV TZ=Europe/Warsaw
 
-# Update and install package dependencies in a single RUN command to reduce layer size
-RUN apt-get update && apt-get full-upgrade -y && \
-    apt-get install --no-install-recommends -y apt-utils sudo build-essential cmake git ca-certificates curl libbsd-dev && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
-
-# Create a non-root user
-ARG USER=docker
-ARG UID=1000
-RUN useradd -l -m -s /bin/bash -u ${UID} ${USER}
-
-# Set default environment variables for the user
+ARG MCM_DIR=/opt/mcm
 ARG PREFIX_DIR="/usr/local"
-ARG HOME="/home/${USER}"
-ENV PATH="${PREFIX_DIR}/bin:$HOME/.local/bin:$HOME/bin:$HOME/usr/bin:$PATH"
 
-# Set the MCM directory
-ARG MCM_DIR=/opt/mcm
+SHELL ["/bin/bash", "-exc"]
+RUN apt-get update && \
+    apt-get full-upgrade -y && \
+    apt-get install --no-install-recommends -y \
+        apt-utils build-essential make cmake git ca-certificates pkg-config nasm libbsd-dev && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
 
-# Clone the MCM repository and build the application
-RUN git clone --depth 1 --branch main https://github.com/OpenVisualCloud/Media-Communications-Mesh.git ${MCM_DIR}
-WORKDIR $MCM_DIR/sdk
+COPY . ${MCM_DIR}
+WORKDIR ${MCM_DIR}/sdk
 RUN ./build.sh
 
-## Re-build container for optimised runtime environment using clean Ubuntu release
-FROM ubuntu:22.04@sha256:2b7412e6465c3c7fc5bb21d3e6f1917c167358449fecac8176c6e496e5c1f05f
-
-# Set the maintainer label
-LABEL maintainer="qiang.han@intel.com"
+ARG IMAGE_CACHE_REGISTRY
+ARG IMAGE_NAME
+FROM ${IMAGE_CACHE_REGISTRY}/${IMAGE_NAME}
 
-# Set environment variables
-ENV DEBIAN_FRONTEND=noninteractive
+LABEL maintainer="milosz.linkiewicz@intel.com"
+LABEL org.opencontainers.image.title="Intel® MCM SDK" \
```

Comment:
Need to use full Media Communication Mesh name instead of MCM

Reply:
Fixed all labels for Docker images.
Added licenses annotations.